graylog2-web-interface/src/views/logic/views/GenerateNextPosition.ts

```diff
@@ -21,14 +21,48 @@ import GetPositionForNewWidget from 'views/logic/views/GetPositionForNewWidget';
 
 import type Widget from '../widgets/Widget';
 
-const incrementRow = (position: WidgetPosition) => position.row + 1;
+export const ConcatPositions = (
+  newPositions: Immutable.Map<string, WidgetPosition>,
+  curPositions: Immutable.Map<string, WidgetPosition>,
+) => {
+  let rowIncrement = 0;
+
+  const newUpdatedPositions = newPositions.map((initialPosition) => {
+    const defaultHeight = initialPosition.height;
+    const row: number = rowIncrement + initialPosition.row;
+    const widgetPosition = initialPosition.toBuilder().row(row).build();
+    rowIncrement += defaultHeight;
+
+    return widgetPosition;
+  }) as Immutable.Map<string, WidgetPosition>;
```

Comment:
You can also use `}).toMap()` here instead of `Immutable.Map<string, WidgetPosition>`